capstone-app/src/app/modules/history/wiki-results.service.ts

```diff
@@ -17,15 +19,74 @@ export class WikiResultsService {
     'https://en.wikipedia.org/w/api.php?origin=*&action=parse&page=';
   private static readonly URL_END = '&format=json';
 
-  search(query: string): Observable<WikiSearchResult> {
+  search(query: string): Observable<{title: string, history: string, furtherReading: Map<string, string>}> {
     return this.searchHandler
       .getQueryString(query)
       .pipe(
         switchMap((result) =>
           this.http.get<WikiSearchResult>(
             `${WikiResultsService.URL_BEGINNING}${result}${WikiResultsService.URL_END}`
           )
-        )
+        ),map(history => {
+          if (history.parse.text) {
+            const historyResult = this.fixString(history.parse.text['*']);
+            const title = history.parse.title;
+            return {title, ...historyResult};
+          }
+          else {
+            throw new Error("API did not return a valid response");
+          }
+        }),
+        catchError(() => {
+          console.log('API did not return a valid response.');
+          const title = '';
+          const history = '';
+          const furtherReading = new Map<string, string>();
+          return of({title, history, furtherReading});
```

Comment:
Since this is for error handling, what's the purpose of returning empty variables? It would probably make more sense to use an error listener when you subscribe to the service and, if necessary, declare them empty in that component rather than adding unnecessary variables to the stack

Reply:
of({title: '', history: '', furtherReading: new Map<string, string>()});